kong/cmd/utils/prefix_handler.lua

```diff
@@ -168,6 +195,19 @@ local function compile_conf(kong_config, conf_template)
     end
   end
 
+  if type(kong_config.lua_ssl_trusted_certificate) == "table" then
+    if #kong_config.lua_ssl_trusted_certificate == 0 then
+      kong_config.lua_ssl_trusted_certificate = nil
+
+    else
+      local filepath, err = gen_trusted_certs_file(kong_config)
+      if not filepath then
+        return nil, err
+      end
+      kong_config.lua_ssl_trusted_certificate = filepath
```

Comment:
Usually when we do config transformations, a new config is generated instead of modifying the existing one. 

For example:

https://github.com/Kong/kong/blob/next/kong/conf_loader/init.lua#L813-L817

I think assigning the filename to a new entry such as `kong_config.lua_ssl_trusted_certificate_combined` follows the convention a bit better.

Reply:
I considered this. Each option has some advantages and some disadvantages.

* Using two different config options, like you are proposing, has the problem that both options ultimately affect the same nginx variable, that is, the nginx `lua_ssl_trusted_certificate` option. I thought this would be ultimately less friendly to the user, because "what happens if I set both options" needs to be answered. Either we "ignore one silently", we "combine both" or we send an error message.
* Extending the existing option means less hussle for the end user. There is some impedance too (what was a 1-to-1 mapping to the nginx config option is now in fact a full-blown Kong option with "magic"). But for all effects and purposes it is backwards-compatible, so users who already used it don't need to change anything, and new users can read the description.

The second option is also slightly more difficult to implement than the first one because it requires "convincing" the config validator into transforming an array into a string, or nil.

> We choose to go to the Moon in this decade and do the other things, not because they are easy, but because they are hard

That said, going around the config validator like this is a bit of a pain and I might just do what you suggested, just to avoid having to do it 😋

Reply:
Update: convincing the conf loader to do this was too difficult. I finally went with your proposed solution (using a separate conf variable).